sitestats/sitestats-impl-hib/src/java/org/sakaiproject/sitestats/impl/SitePresenceTotalImpl.java

```diff
@@ -33,11 +36,21 @@
 
     private static final long serialVersionUID = 1L;
 
+    private static final Log LOG = LogFactory.getLog(SitePresenceTotalImpl.class);
+
     private long id;
     private String siteId;
     private String userId;
     private int totalVisits;
+
     private Date lastVisitTime;
+    public void setLastVisitTime(Date lastVisitTime) {
+        if (lastVisitTime == null) {
+            LOG.error("lastVisitTime is null. Setting to current date ...");
```

Comment:
I don't know if it is expected. Why should a presence object ever lack a timestamp?